### PR TITLE
Adds support to update and delete a server profile template

### PR DESCRIPTION
**Pull request by bsouza**

### CODE REVIEW

File: examples/server_profile_templates.py [hunk 1]
```diff
@@ -33,13 +33,37 @@
...
+basic_template_options = dict(
+    name=server_profile_name,
+    serverHardwareTypeUri=server_hardware_type_uri,
+    enclosureGroupUri=enclosure_group_uri
+)
+basic_template = oneview_client.server_profile_templates.create(basic_template_options)
+pprint(basic_template)
+
+# Update hideUnusedFlexNics from recently created template
+print("\nUpdate the")
```
Comment (marikrg) [inline examples/server_profile_templates.py]: Isn't this sentence incomplete?

File: hpOneView/resources/servers/server_profile_templates.py [hunk 2]
```diff
@@ -87,3 +90,80 @@ def get(self, id_or_uri):
...
+        simple as a name or description change or much more complex changes around the networking configuration.
+        It should be noted that selection of a virtual or physical MAC or Serial Number type is not mutable once a
+        profile template has been created, and attempts to change those elements will not be applied to the target
+        profile template. Connection requests can be one of the following types - port Auto, auto and explicit.
+        An explicit request is where the request portId parameter includes the adapter, port and flexNic. An auto
+        request  is where portId is set to "Auto" and a port auto request is where just the portId parameter includes
+        just the adapter and port. The fields listed as required in the Request Body section need to be specified
+        only when their associated parent is used.
+
+        Returns:
```
Comment (marikrg) [inline hpOneView/resources/servers/server_profile_templates.py]: You forgot the Args section.

File: examples/server_profile_templates.py (hunk 1 above)
Comment (bsouza) [inline examples/server_profile_templates.py]: Done!

File: hpOneView/resources/servers/server_profile_templates.py (hunk 2 above)
Comment (bsouza) [inline hpOneView/resources/servers/server_profile_templates.py]: Done!

**Comment by piantola** [comment]
:+1:

**Pull request merged**